prescient.el

```diff
@@ -79,15 +79,14 @@ This only has an effect if `prescient-persist-mode' is enabled."
   :type 'file)
 
 (defvar prescient--filter-method-custom-type
-  '(choice
+  '(set
     (const :tag "Literal" literal)
-    (const :tag "Initialism" initialism)
-    (const :tag "Literal and initialism" literal+initialism)
```

Comment:
I agree with your assessment that removing `literal+initialism` is unlikely to cause problems. However, I'd like to provide a stable API even at the cost of a little inconvenience. (Any breaking change will need to be documented in the changelog.)

How about this: we leave `prescient--filter-method-custom-type` as you have it, but in `prescient-filter-regexps`, we add a third case to your current `if` statement to account for the special value?

The fact that this symbol is allowed for backwards compatibility should then be documented in `prescient-filter-method`.

(I take this approach [here](https://github.com/raxod502/straight.el/blob/e17e54790ca6f29afb17d89f48569f03074ff682/straight.el#L1023-L1040) and [here](https://github.com/raxod502/straight.el/blob/e17e54790ca6f29afb17d89f48569f03074ff682/straight.el#L2840-L2875) in `straight.el`.)